lib/tasks/db/update_annotation_line.rake

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+namespace :db do
+  task update_annotation_line: :environment do
+    Rails.application.eager_load!
+
+    require 'action_view'
+    require 'action_view/helpers'
+
+    helper = HelperContext.new
+
+    start_date_input, end_date_input = ENV['START_DATE'], ENV['END_DATE']
+    unless start_date_input && end_date_input
+      puts 'Please provide START_DATE and END_DATE to execute this script'
+      exit
+    end
+
+    start_date = Date.parse(start_date_input)
+    end_date = Date.parse(end_date_input)
+
+    if start_date > end_date
+      puts 'START_DATE should be earlier than END_DATE'
+      exit
+    end
+
+    start_time = Time.now
+
+    ActsAsTenant.without_tenant do
+      conn = ActiveRecord::Base.connection
+      annotations_with_file_content = conn.exec_query(<<-SQL)
+        SELECT annotations.*, files.content, questions.actable_id
+        FROM course_assessment_answer_programming_file_annotations AS annotations
+        JOIN course_assessment_answer_programming_files AS files
+          ON annotations.file_id = files.id
+        JOIN course_assessment_answers AS answers
+          ON files.answer_id = answers.actable_id
+          AND answers.actable_type = 'Course::Assessment::Answer::Programming'
+          AND CAST(answers.created_at AS DATE) BETWEEN '#{start_date}' AND '#{end_date}'
+        JOIN course_assessment_questions AS questions
+          ON answers.question_id = questions.id
+      SQL
+
+      end_time = Time.now
+      puts "Extract all annotations with file content = #{end_time - start_time} seconds"
+
+      annotations_grouped = annotations_with_file_content.group_by do |row|
```

Comment:
Why is this outside of time profiling? How long does the `.group_by do` take?

Reply:
sorry, I was not thinking it would be relevant to our case, since this technically is just grouping the already extracted data from DB so that the processing for the highlighter might be more efficient..

I was including the profiling for this command, and running from `2021-07-01` to `2022-01-01`, it takes 0.06 seconds.